src/extension/csw/simple-store/src/test/java/org/geoserver/csw/store/simple/SimpleCatalogStoreTest.java

```diff
@@ -72,22 +73,11 @@ public void testCapabilities() throws Exception {
     @Test
     public void testCreationExceptions() throws IOException {
         try {
-            new SimpleCatalogStore(new File("./pom.xml"));
+            new SimpleCatalogStore(Files.asResource(new File("./pom.xml")));
             fail("Should have failed, the reference is not a directory");
         } catch(IllegalArgumentException e) {
             // fine
         }
-        
-        File f = new File("./target/notThere");
-        if(f.exists()) {
-            FileUtils.deleteDirectory(f);
-        }
-        try {
-            new SimpleCatalogStore(f);
-            fail("Should have failed, the reference is not there!");
-        } catch(IllegalArgumentException e) {
-            // fine
-        }
```

Comment:
@bencaradocdavies No, this is part of the change in 'philosophy' when it comes to the creation of directories. See the discussion I had with @jodygarnett a few months ago on the ML:

* In the resource store API, there is no way to create a resource directory in the store. They are created automatically on the fly as you create a resource for the first time inside that directory (possibly even multiple directories on the path of a resource do not exist yet - you can create children of non-existing resources). 
As a consequence, whenever a Resource variable that holds a directory is used to read and/or write multiple resources, there is never a checked whether it exists or not. It is simply instantiated and used as if it already exists.

* Previously, The SimpleCatalogStore threw an error if a non-existing directory was passed to it. Its child class, GeoServerSimpleCatalogStore would create the "$DATA_DIR/catalog" if necessary and pass it on.
This is not possible any more for the reason explained above.

As a consquence, I was forced to remove the exception that was thrown when the directory does not exist yet, and thus also the test for this behaviour.